addons/cookoff/functions/fnc_engineFireLocal.sqf

```diff
@@ -27,7 +27,11 @@ if (hasInterface) then {
     private _hitPoints = getAllHitPointsDamage _vehicle;
 
     // Get hitpoint for engine
-    private _index = (_hitPoints select 0) findIf {_x == "hitengine"};
+    private _index = if (_hitPoints isNotEqualTo []) then {
+        (_hitPoints select 0) findIf {_x == "hitengine"}
+    } else {
+        -1
```

Comment:
```suggestion
    private _index = if (_hitPoints isEqualTo []) then {
        -1
    } else {
        (_hitPoints select 0) findIf {_x == "hitengine"}
```

Reply:
I don't like that the "heavier" amount of code is in the `else` block. Imo the more important stuff should be in the `if` block.